evaluator.go

```diff
@@ -527,13 +570,17 @@ func (n nodeFunction) visit(state *State, context *evalContext) error {
 
 		lookingFor := subContext.currentResult
 
-		for clusterName, _ := range state.clusters {
-			subContext = context.subCluster(clusterName)
-			clusterLookup(state, &subContext, "CLUSTER")
-
-			for value := range subContext.resultIter() {
-				if lookingFor.Contains(value) {
-					context.addResult(clusterName)
+		if state.cacheClusterQueryResults {
+			results := state.getResultsFromCachedCQRforSlice(lookingFor.ToSlice())
+			context.addSliceToResult(results.ToSlice())
```

Comment:
Using slices feels weird to me, mostly because we don't have to do it anywhere else. Why is it necessary here? (Rather than just using set operations used elsewhere?)